source/opt/decompose_initialized_variables_pass.cpp

```diff
@@ -57,20 +72,41 @@ Pass::Status DecomposeInitializedVariablesPass::Process() {
       Instruction* store_inst = new Instruction(
           context(), SpvOpStore, 0, 0,
           {{SPV_OPERAND_TYPE_ID, {var_id}}, {SPV_OPERAND_TYPE_ID, {val_id}}});
-      new_stores.push_back(store_inst);
+      function_stores.push_back(store_inst);
       iter->RemoveOperand(3);
```

Comment:
Does this mess up the def-use manager as well?

Reply:
Yup, fixed it.